Sources/ProjectSpec/AggregateTarget.swift

```diff
@@ -62,6 +62,35 @@ extension AggregateTarget: NamedJSONDictionaryConvertible {
     }
 }
 
+extension AggregateTarget: JSONDictionaryEncodable {
+    public func toJSONDictionary() -> JSONDictionary {
+        var dict: JSONDictionary = [:]
+
+        let settingsDict = settings.toJSONDictionary()
+        if settingsDict.count > 0 {
+            dict["settings"] = settingsDict
+        }
+        
+        if targets.count > 0 {
+            dict["targets"] = targets
```

Comment:
All these `if array.count > 0 {` are not really required. It might be useful for some use cases to generate out all the empty arrays. For the compact spec you have now, there could be a function `JSONDictionary.removeEmptyArrays` that loops through and removes all empty arrays

Reply:
Definitively a good approach generating empty template really help getting started.